crates/wasm-encoder/src/code.rs

```diff
@@ -150,7 +150,7 @@ impl Function {
 #[derive(Clone, Copy, Debug)]
 pub struct MemArg {
     /// A static offset to add to the instruction's dynamic address operand.
-    pub offset: u32,
+    pub offset: u64,
```

Comment:
Should also update this field's doc comment to talk about when one can use offsets > u32::MAX